### PR TITLE
Enable Translation into other languages

### PR DESCRIPTION
**Pull request by corsacca**
image

### CODE REVIEW

**Comment by corsacca** [comment]
See https://github.com/Pray4Movement/prayer-global-porch/issues?q=is%3Aissue+is%3Aopen+Translation+%3E
For remaining places that need to be fixed for translation

**Review by squigglybob — CHANGES_REQUESTED** [review]
* The un hovered language in the selector is white text.
* clicking on french goes to the re-auth screen
https://www.loom.com/share/daad39eb410946e7adeb6cb9937c998d?sid=48fd71f0-0f7e-4930-bfd3-fa32db22fd05

* share.js l15 needs translating for the share functionality
* as does 'Done' in the share modal 
image
* heatmap.js line ~1000 there are a few strings there (seen on the race map when clicking 
 on a state)
* race-list.js I totally hadn't done at all!! sorry 🙁
* and jquery DataTable would need localisation too 😭 '/race_app/race_list/' in active-list.js and race-list.js
* we should probably make these have the same capitalization 
image
* I don't think this is for now, but I wonder if we can get translated place names from the mapbox API 
image
* in managing prayer relays in the profile area, there are some missing translations
image
* what are your thoughts on translating the webform on the contacts page? I think we talked once about maybe having the form just in PG rather than imported from somewhere else, that would certainly make translsating it easier  (and styling it to match the site too).
image
* just remembered we had a magic-privacy page, i've wrapped them in gettext, but will need translating
* likewise with the 404.php page

I'm loving the translateability and the selector in the nav bar. Can't wait to see it ready and released.
I'm happy to jump over a bit to help with any of these issues when I can.
I also leave the above up to you whether they need to be in this PR or punted to another issue

File: utilities/global-utilities.php [hunk 1]
```diff
@@ -1025,3 +1025,65 @@ function dt_sanitize_array( &$array ) {
...
+ *
+ * @return string
+ */
+function pg_get_current_lang(): string {
+    $lang = 'en_US';
+    if ( defined( 'PORCH_DEFAULT_LANGUAGE' ) ){
+        $lang = PORCH_DEFAULT_LANGUAGE;
+    }
+    if ( isset( $_GET['lang'] ) && !empty( $_GET['lang'] ) ){
+        $lang = sanitize_text_field( wp_unslash( $_GET['lang'] ) );
```
[the remark this answers is not in the document]
Comment (corsacca) [reply utilities/global-utilities.php:1044]: The current strategy might not even need the ?lang=fr. That is useful in links handed out. The lang is set in the cookie and the site feels native.
/fr/ strategy requires all the pages all the time to have that added to the path. And makes some functions harder like  `dt_url_path()`

File: pages/user/user-profile.php [hunk 2]
```diff
@@ -61,60 +61,63 @@ public function dt_magic_url_base_allowed_js( $allowed_js ) {
...
+            'user-profile-js'
         ]);
     }
 
-    public function wp_enqueue_scripts() {}
+    public function wp_enqueue_scripts() {
+        pg_enqueue_components();
+        wp_enqueue_script( 'user-profile-js', trailingslashit( plugin_dir_url( __FILE__ ) ) . 'user-profile.js', [ 'jquery', 'components-js' ], filemtime( trailingslashit( plugin_dir_path( __FILE__ ) ) . 'user-profile.js' ), true );
+        wp_localize_script( 'user-profile-js', 'jsObject', [
+            'map_key' => DT_Mapbox_API::get_key(),
```
[the remark this answers is not in the document]
Comment (corsacca) [reply pages/user/user-profile.php:73]: Yeah, I'd like to move all the enqueues to uses this strategy

File: pages/pray/lap.js [hunk 3]
```diff
@@ -1,3 +1,19 @@
...
+  if (typeof str === "undefined") return '';
+  if (typeof str !== "string") return str;
+  return str.replace(/&/g, "&amp;").replace(/</g, "&lt;").replace(/>/g, "&gt;").replace(/"/g, "&quot;").replace(/'/g, "&apos;");
+};
+const translations = escapeObject(jsObject.translations)
+let translate = function ( string ){
+  return translations[string] ? translations[string] : string
+}
+
+
```
[the remark this answers is not in the document]
Comment (corsacca) [reply pages/pray/lap.js:16]: We are not loading D.T's shared-functions.js on all the P.G front end pages. 
But I'll move these to P.G's global functions to be used throughout.
The translate function does depend on the page's localized object.

File: utilities/global-utilities.php (hunk 1 above)
[the remark this answers is not in the document]
Comment (squigglybob) [reply utilities/global-utilities.php:1044]: ah yeah I like that better, let's keep it that way.

**Comment by corsacca** [comment]
After a bunch of changes
- all the english text i can find is waiting on the changes to be pushed to weblate for translation
- we won't translate the webform, that will require a bunch of work on the webform plugin.
- the remaining translation issues are logged here: https://github.com/Pray4Movement/prayer-global-porch/issues?q=is%3Aissue+is%3Aopen+label%3Atranslation

**Review by squigglybob — CHANGES_REQUESTED** [review]
It's all looking good. I like the idea of using the bootsrap  dropdown, it looks nicer and solves some issues.
I changed up the styles slightly and switched out the icon.

* on the page /race_app/race_map/
there might be a duplicated heatmap.js declaration going  on
image
* and on the active  relays page and the lap.js page
image
* wierdly when I click on status link in menu when not on the home page it works fine, but when on the home page it has a jquery-ui missing error
image

I'm guessing something is up with the script loadings, but other than that it's all looking good bro 😃

**Comment by squigglybob** [comment]
looks good bro thanks 😃 🎉

**Pull request merged**